crates/common/src/opa.rs

```diff
@@ -280,6 +292,104 @@ impl PolicyLoader for CliPolicyLoader {
     }
 }
 
+#[derive(Clone, Default)]
+pub struct UrlPolicyLoader {
+    policy_id: String,
+    address: String,
+    policy: Vec<u8>,
+    entrypoint: String,
+}
+
+impl UrlPolicyLoader {
+    pub fn new(url: &str, policy_id: &str, entrypoint: &str) -> Self {
+        Self {
+            address: url.into(),
+            policy_id: policy_id.to_owned(),
+            entrypoint: entrypoint.to_owned(),
+            ..Default::default()
+        }
+    }
+
+    async fn fetch_bundle_from_url(&mut self) -> Result<Vec<u8>, PolicyLoaderError> {
+        let response = reqwest::get(&self.address).await?;
+        if response.status().is_success() {
+            let bytes = response.bytes().await?;
+            Ok(bytes.to_vec())
+        } else {
+            Err(PolicyLoaderError::HttpError(response.status().to_string()))
+        }
+    }
+}
+
+#[async_trait::async_trait]
+impl PolicyLoader for UrlPolicyLoader {
+    fn set_address(&mut self, address: &str) {
+        self.address = address.to_owned();
+    }
+
+    fn set_rule_name(&mut self, name: &str) {
+        self.policy_id = name.to_owned();
+    }
+
+    fn set_entrypoint(&mut self, entrypoint: &str) {
+        self.entrypoint = entrypoint.to_owned();
+    }
+
+    fn get_address(&self) -> &str {
+        &self.address
+    }
+
+    fn get_rule_name(&self) -> &str {
+        &self.policy_id
+    }
+
+    fn get_entrypoint(&self) -> &str {
+        &self.entrypoint
+    }
+
+    fn get_policy(&self) -> &[u8] {
+        &self.policy
+    }
+
+    fn load_policy_from_bytes(&mut self, policy: &[u8]) {
+        self.policy = policy.to_vec();
+    }
+
+    async fn load_policy(&mut self) -> Result<(), PolicyLoaderError> {
+        let url = &self.address;
+
+        let bundle = if url.starts_with("http://")
```

Comment:
No need to repeat parsing logic when we already have it. (Also, what of "HTTP://" etc.?) Just see if `Url::parse` likes it or not and, if it doesn't, fall back to parsing as `Path`?

Reply:
Can we do another round of review on this bit, please? Not there yet, I don't think, but a bit closer.

Reply:
Sure thing, will take another look today.